MLIR.cmake

```diff
@@ -52,300 +55,59 @@ message(STATUS "LLVM_PROJECT_LIB        : " ${LLVM_PROJECT_LIB})
-        ${MLIRStandard}
-        ${MLIRStandardOpsTransforms}
-        ${MLIRStandardToLLVM}
-        ${MLIRTransforms}
-        ${MLIRSCFToStandard}
-        ${MLIRVector}
-        ${MLIRVectorInterfaces}
-        ${MLIRVectorToLLVM}
-        ${MLIRVectorToSCF}
-        ${MLIRSCF}
-        ${MLIRIR}
-        ${MLIRLLVMIR}
-        ${MLIROptLib}
-        ${MLIRParser}
-        ${MLIRPass}
-        ${MLIRTargetLLVMIR}
-        ${MLIRTargetLLVMIRModuleTranslation}
-        ${MLIRTransforms}
-        ${MLIRTransformUtils}
-        ${MLIRAffine}
-        ${MLIRAffineToStandard}
-        ${MLIRAffineTransforms}
-        ${MLIRAnalysis}
-        ${MLIRCallInterfaces}
-        ${MLIRControlFlowInterfaces}
-        ${MLIRDialect}
-        ${MLIREDSC}
-        ${MLIRExecutionEngine}
-        ${MLIRIR}
-        ${MLIRLLVMIRTransforms}
-        ${MLIRSCFToStandard}
-        ${MLIRSCF}
-        ${MLIRSCFTransforms}
-        ${MLIRLoopAnalysis}
-        ${MLIRLoopLikeInterface}
-        ${MLIROpenMP}
-        ${MLIRMlirOptMain}
-        ${MLIRSideEffectInterfaces}
-        ${MLIRStandard}
-        ${MLIRStandardToLLVM}
-        ${MLIRTranslation}
-        ${MLIRSupport}
-        ${MLIRLinalg}
-        ${MLIRLinalgEDSC}
-        ${MLIRLinalgAnalysis}
-        ${MLIRLinalgTransforms}
-        ${MLIRLinalgUtils}
-        ${MLIRAffineEDSC}
-        ${MLIRLinalgEDSC}
-        ${MLIRViewLikeInterface}
-        ${MLIRPresburger}
-        ${MLIRRewrite}
-        ${MLIRShape}
-        ${MLIRShapeToStandard}
-        ${MLIRInferTypeOpInterface}
-        ${MLIRRewrite}
-        ${MLIRAnalysis}
-        ${MLIRPDLInterp}
-        ${MLIRPDLToPDLInterp}
-        ${MLIRPDL}
-        ${MLIRTensor}
-        ${MLIRArmNeonToLLVM}
-        ${MLIRLLVMArmNeon}
-        ${MLIRArmNeon}
-        ${MLIRArmSVEToLLVM}
-        ${MLIRLLVMArmSVE}
-        ${MLIRArmSVE}
-        # strict order verified
-        ${LLVMBitWriter}
-        ${LLVMObject}
-        ${LLVMBitReader}
-        # strict order verified
-        ${LLVMFrontendOpenMP}
-        ${LLVMTransformUtils}
-        ${LLVMAnalysis}
-        # strict order verified
-        ${LLVMAsmParser}
-        ${LLVMCore}
-        # strict order not yet verified
-        ${LLVMRemarks}
-        ${LLVMMCParser}
-        ${LLVMMC}
-        ${LLVMProfileData}
-        ${LLVMBinaryFormat}
-        ${LLVMBitstreamReader}
-        ${LLVMIRReader}
-        ${LLVMMLIRTableGen}
-        ${LLVMSupport}
-        ${LLVMDemangle}
-        ${MLIR_SYSTEM_LIBS})
-
-if (USE_INSTALLED_LLVM)
-  set(LLVM_CMAKE_DIR
-        "${LLVM_PROJECT_LIB}/cmake/llvm"
-        CACHE PATH "Path to LLVM cmake modules")
-else()
-  set(LLVM_CMAKE_DIR
-        "${LLVM_PROJ_BUILD}/lib/cmake/llvm"
-        CACHE PATH "Path to LLVM cmake modules")
-endif()
-list(APPEND CMAKE_MODULE_PATH "${LLVM_CMAKE_DIR}")
-include(AddLLVM)
-include(TableGen)
+# FIXME: Remove after migration
+#set(MLIRLibs ${MLIR_ALL_LIBS})
+get_property(MLIRLibs GLOBAL PROPERTY MLIR_ALL_LIBS)
```

Comment:
Not all of these libs are needed, you can do:

  get_property(dialect_libs GLOBAL PROPERTY MLIR_DIALECT_LIBS)
  get_property(conversion_libs GLOBAL PROPERTY MLIR_CONVERSION_LIBS)

to limit the list